--- unifdefall.sh.orig	2009-06-27 12:57:29.000000000 +0800
+++ unifdefall.sh	2009-06-29 01:43:05.000000000 +0800
@@ -8,22 +8,22 @@
 set -e
 
 basename=`basename $0`
-tmp=`mktemp -d -t $basename` || exit 2
+tmp=`mktemp -d -t $basename.XXX` || exit 2
 
 unifdef -s "$@" | sort | uniq > $tmp/ctrl
 cpp -dM "$@" | sort |
-	sed -Ee 's/^#define[ 	]+(.*[^	 ])[ 	]*$/\1/' > $tmp/hashdefs
-sed -Ee 's/^([A-Za-z0-9_]+).*$/\1/' $tmp/hashdefs > $tmp/alldef
+	sed -re 's/^#define[ 	]+(.*[^	 ])[ 	]*$/\1/' > $tmp/hashdefs
+sed -re 's/^([A-Za-z0-9_]+).*$/\1/' $tmp/hashdefs > $tmp/alldef
 comm -23 $tmp/ctrl $tmp/alldef > $tmp/undef
 comm -12 $tmp/ctrl $tmp/alldef > $tmp/def
 
 echo unifdef -k \\ > $tmp/cmd
-sed -Ee 's/^(.*)$/-U\1 \\/' $tmp/undef >> $tmp/cmd
+sed -re 's/^(.*)$/-U\1 \\/' $tmp/undef >> $tmp/cmd
 while read sym
-do	sed -Ee '/^('"$sym"')([(][^)]*[)])?([ 	]+(.*))?$/!d;s//-D\1=\4/' $tmp/hashdefs
+do	sed -re '/^('"$sym"')([(][^)]*[)])?([ 	]+(.*))?$/!d;s//-D\1=\4/' $tmp/hashdefs
 done < $tmp/def |
-	sed -Ee 's/\\/\\\\/g;s/"/\\"/g;s/^/"/;s/$/" \\/' >> $tmp/cmd
+	sed -re 's/\\/\\\\/g;s/"/\\"/g;s/^/"/;s/$/" \\/' >> $tmp/cmd
 echo '"$@"' >> $tmp/cmd
-sh $tmp/cmd "$@"
+sh $tmp/cmd "$@" || test $? = 1
 
 rm -r $tmp
