diff -urN ussp-push-0.11/src/obex_main.c ussp-push-0.11.new/src/obex_main.c
--- ussp-push-0.11/src/obex_main.c	2008-08-24 13:48:53.000000000 -0700
+++ ussp-push-0.11.new/src/obex_main.c	2018-03-24 17:07:19.000000000 -0700
@@ -62,6 +62,8 @@
 #include "obex_main.h"
 #include "obex_socket.h"
 
+#include "openobex_utils.h"
+
 #define UPUSH_APPNAME "ussp-push v0.11"
 #define BT_SERVICE "OBEX"
 #define OBEX_PUSH        5
@@ -197,7 +199,7 @@
 		*err = -1;
 		return NULL;
 	}
-	if (!(handle = OBEX_Init(OBEX_TRANS_CUST, obex_event, 0))) {
+	if (!(handle = OBEX_Init(OBEX_TRANS_CUSTOM, obex_event, 0))) {
 		BTERROR("OBEX_Init failed: %s\n", strerror(errno));
 		obex_free(gt);
 		*err = -1;
diff -urN ussp-push-0.11/src/openobex_utils.h ussp-push-0.11.new/src/openobex_utils.h
--- ussp-push-0.11/src/openobex_utils.h	1969-12-31 16:00:00.000000000 -0800
+++ ussp-push-0.11.new/src/openobex_utils.h	2018-03-24 17:06:58.000000000 -0700
@@ -0,0 +1,48 @@
+/**
+ * Copied from openobex-1.7.1/apps/lib/obex_io.c
+ * These functions are not exported by libopenobex.so any more with unknown reason.
+ */
+
+static inline
+int OBEX_UnicodeToChar(uint8_t *c, const uint8_t *uc, int size)
+{
+	int n;
+
+	if (uc == NULL || c == NULL)
+		return -1;
+
+	/* Make sure buffer is big enough! */
+	for (n = 0; uc[n*2+1] != 0; n++);
+
+	if (n >= size)
+		return -1;
+
+	for (n = 0; uc[n*2+1] != 0; n++)
+		c[n] = uc[n*2+1];
+	c[n] = 0;
+
+	return 0;
+}
+
+static inline
+int OBEX_CharToUnicode(uint8_t *uc, const uint8_t *c, int size)
+{
+	int len, n;
+
+	if (uc == NULL || c == NULL)
+		return -1;
+
+	len = n = strlen((char *) c);
+	if (n*2+2 > size)
+		return -1;
+
+	uc[n*2+1] = 0;
+	uc[n*2] = 0;
+
+	while (n--) {
+		uc[n*2+1] = c[n];
+		uc[n*2] = 0;
+	}
+
+	return (len * 2) + 2;
+}
