diff -ru xf86-video-intel-2.9.1/uxa/uxa-render.c xf86-video-intel-2.9.1.new/uxa/uxa-render.c
--- xf86-video-intel-2.9.1/uxa/uxa-render.c	2009-10-26 19:48:05.000000000 +0800
+++ xf86-video-intel-2.9.1.new/uxa/uxa-render.c	2010-08-24 10:54:19.000000000 +0800
@@ -440,6 +440,8 @@
 
 #if XORG_VERSION_CURRENT < XORG_VERSION_NUMERIC (1,6,99,1,0)
 #define IMAGE_FROM_PICT(P) image_from_pict(P, 0)
+#elif XORG_VERSION_CURRENT > XORG_VERSION_NUMERIC (1,8,0,0,0)
+#define IMAGE_FROM_PICT(P) image_from_pict(P, 0, 0, 0)
 #else
 #define IMAGE_FROM_PICT(P) image_from_pict(P, 0, 0)
 #endif
