--- configure.old	2011-06-21 00:01:06.000000000 +0800
+++ configure	2011-07-20 13:25:15.000000000 +0800
@@ -31499,7 +31499,7 @@
 		for file in `ldconfig -p | grep /libc.so| xargs -n 1 echo | grep /libc.so`; do
 		    test -s "$file" && LIBC_ALTS="${LIBC_ALTS}${LIBC_ALTS:+ }$file"
 		done
-		LIBC_NAME=`echo ${LIBC_ALTS} | sed -e 's/.*\///' | sort -nr | head -n 1`
+		LIBC_NAME=`echo ${LIBC_ALTS} | sed -e 's/ \+/\n/g' | sed -e 's/.*\///' | sort -nr | head -n 1`
 		if test "x${LIBC_NAME}" = x; then
 			#nothing found, set libc.so anyway
 			LIBC_NAME="${base_library_path}libc.so"
@@ -31779,6 +31779,8 @@
 int
 main(int argc, char *argv[])
 {
+   return 1;	/* In Linux system, there is no program named 'lari',
+   		   so return 1 directly. (means preload can be enabled) */
    char buf[1024];
 
    strcpy(buf, "lari -V ");
