diff -ur giflib-4.2.3/lib/gif_err.c giflib-4.2.3.new/lib/gif_err.c
--- giflib-4.2.3/lib/gif_err.c	2013-09-03 21:41:31.000000000 +0800
+++ giflib-4.2.3.new/lib/gif_err.c	2014-08-09 15:35:48.000000000 +0800
@@ -119,3 +119,13 @@
     }
     return Err;
 }
+
+void
+PrintGifError(void) {
+    char *Err = GifErrorString();
+
+    if (Err != NULL)
+        fprintf(stderr, "\nGIF-LIB error: %s.\n", Err);
+    else
+        fprintf(stderr, "\nGIF-LIB undefined error %d.\n", GifError());
+}
diff -ur giflib-4.2.3/util/qprintf.c giflib-4.2.3.new/util/qprintf.c
--- giflib-4.2.3/util/qprintf.c	2013-09-03 21:41:31.000000000 +0800
+++ giflib-4.2.3.new/util/qprintf.c	2014-08-09 15:35:43.000000000 +0800
@@ -36,13 +36,3 @@
 
     va_end(ArgPtr);
 }
-
-void
-PrintGifError(void) {
-    char *Err = GifErrorString();
-
-    if (Err != NULL)
-        fprintf(stderr, "\nGIF-LIB error: %s.\n", Err);
-    else
-        fprintf(stderr, "\nGIF-LIB undefined error %d.\n", GifError());
-}
