diff -ru openvanilla-modules-0.8.0_12/OVIMGeneric/OVIMGeneric.cpp openvanilla-modules-0.8.0_12.new/OVIMGeneric/OVIMGeneric.cpp
--- openvanilla-modules-0.8.0_12/OVIMGeneric/OVIMGeneric.cpp	2009-03-26 15:45:19.000000000 +0800
+++ openvanilla-modules-0.8.0_12.new/OVIMGeneric/OVIMGeneric.cpp	2009-10-09 01:10:12.000000000 +0800
@@ -32,6 +32,7 @@
 
 #include "OVIMGeneric.h"
 #include "CIN-Defaults.h"
+#include <stdio.h>
 
 using namespace std;
 
diff -ru openvanilla-modules-0.8.0_12/SharedSource/OVCINInfo.cpp openvanilla-modules-0.8.0_12.new/SharedSource/OVCINInfo.cpp
--- openvanilla-modules-0.8.0_12/SharedSource/OVCINInfo.cpp	2009-03-01 20:18:55.000000000 +0800
+++ openvanilla-modules-0.8.0_12.new/SharedSource/OVCINInfo.cpp	2009-10-09 01:10:30.000000000 +0800
@@ -31,6 +31,7 @@
 // #define OV_DEBUG
 
 #include "OVCINInfo.h"
+#include <stdio.h>
 
 #ifndef WIN32
     #include <OpenVanilla/OpenVanilla.h>
diff -ru openvanilla-modules-0.8.0_12/SharedSource/OVCandidateList.cpp openvanilla-modules-0.8.0_12.new/SharedSource/OVCandidateList.cpp
--- openvanilla-modules-0.8.0_12/SharedSource/OVCandidateList.cpp	2008-08-06 14:00:25.000000000 +0800
+++ openvanilla-modules-0.8.0_12.new/SharedSource/OVCandidateList.cpp	2009-10-09 01:10:21.000000000 +0800
@@ -31,6 +31,7 @@
 #include "OVCandidateList.h"
 #include "OVCIN.h"
 #include <iostream>
+#include <stdio.h>
 
 #ifdef __linux__
 	#include <string.h>
